### PR TITLE
Use prettier

### PR DESCRIPTION
**Pull request by tschaub**
I've been enjoying using Prettier on other projects for a few years.  Although initially the style choices may not conform with everyone's tastes, after a while I've found that I no longer notice and instead enjoy having the machine make all whitespace choices for me.

I'm opening this pull request to get the opinions of others before proposing that we update the configuration for OpenLayers.  The change there will involve a ton of churn, and it will break all outstanding pull requests.  After that one time pain, things will work as they do currently – no extra tools or editor configuration is needed on the part of contributors – in particular, people will not have to and _should not_ configure their editors to use Prettier directly.

The reason it is possible to use Prettier with OpenLayers now is because it now accepts JSDoc style type casts (see prettier/prettier#7791).

### CODE REVIEW

File: examples/base/main.js [hunk 1]
```diff
@@ -3,30 +3,35 @@ import path from 'path';
...
-  return Promise.all(paths.map(function(name) {
-    return new Promise(function(resolve, reject) {
-      fs.readFile(name, function(err, data) {
-        if (err) {
-          reject(err);
-        } else {
-          resolve(String(data));
-        }
+  return Promise.all(
+    paths.map((name) => {
```
Comment (tschaub) [inline examples/base/main.js:7]: We don't have to use arrow functions, but I wanted some examples of formatting where we do use them.

File: examples/base/main.js [hunk 2]
```diff
@@ -3,30 +3,35 @@ import path from 'path';
...
 if (require.main === module) {
   process.stdout.write(foo());
 
-  const paths = ['main.js', '.eslintrc'].map(function(name) {
-    return path.join(__dirname, name);
-  });
-  main(paths).then(function(contents) {
-    contents.forEach(function(content, index) {
-      process.stdout.write(paths[index] + '\n' + content + '\n');
+  const paths = /** @type {Array<string>} */ (['main.js', '.eslintrc']);
```
Comment (tschaub) [inline examples/base/main.js:24]: This is here to demonstrate that JSDoc type casts are supported.

File: examples/base/mod.js [hunk 3]
```diff
@@ -1,3 +1,10 @@
+const obj = {
+  'bar': 'foo',
+};
+
 export function foo() {
-  return 'foo';
+  // This is a legacy style from a time when Closure Compiler would mangle property names.
+  // TODO: Confirm this is still needed and remove the `quoteProps: 'preserve'`
+  // option for prettier if not.
+  return obj['bar'];
```
Comment (tschaub) [inline examples/base/mod.js:9]: I think we have determined that quoting object properties is no longer necessary.  I don't think we need to change our convention now, but if we want to in the future, we can remove the `quoteProps: 'preserve'` configuration.

File: examples/react/main.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ class Widget extends Component {
 }
 
 Widget.propTypes = {
-  name: string.isRequired
+  name: string.isRequired,
```
Comment (tschaub) [inline examples/react/main.js:11]: If we don't want to have trailing commas, we can configure the Prettier rule with `trailingComma: 'none'`.  I thought I'd start out by enforcing trailing commas though.  I've come to appreciate how much easier it is to edit multi-line object literals and arrays when they are used.

**Comment by tonio** [comment]
We use it internally too. Like it pretty much.

+1 from me.

File: examples/react/main.js (hunk 4 above)
Comment (ahocevar) [reply examples/react/main.js:11]: Do our transpile steps properly remove trailing commas? If not, enforcing leading commas would also be an option. But I'm fine with either.

**Comment by ahocevar** [comment]
Very nice! I've been using Prettier in other projects as well.

**Comment by tschaub** [comment]
See openlayers/openlayers#10862 for what this will look like in OpenLayers.

**Pull request merged**